src/cache.c

```diff
@@ -144,6 +145,24 @@ void CacheSystem_init(const char *path, int url_supplied)
     CACHE_SYSTEM_INIT = 1;
 }
 
+#define NA __attribute__ ((unused))
+static int ntfw_cb(const char *fpath, NA const struct stat *sb, NA int typeflag, NA struct FTW *ftwbuf)
+{
+    return remove(fpath);
+}
+
+void CacheSystem_clear(const char *path, int url_supplied)
+{
+    if (url_supplied) {
+        path = CacheSystem_calc_dir(path);
+    }
+
+    lprintf(debug, "%s\n", path);
+
+    nftw(path, ntfw_cb, 64, FTW_DEPTH | FTW_PHYS | FTW_MOUNT);
```

Comment:
I don't understand why you have to add that extra function.

Reply:
`nftw` and other `fts_*` wrappers seemed to be the way to walk directories and delete them in C. It seemed like a lot to put the entire function in the case-27 block.